File: server/internal/instance/video_apps.go
package instance

import "fmt"

// VideoApps names supported iPhone/iPad clients, never arbitrary launch URLs.
// Empty values inherit instance defaults, which use the service's official app.
type VideoApps struct {
	IOS string `json:"ios"`
}

func IsVideoServerType(serviceType string) bool {
	return serviceType == "plex" || serviceType == "jellyfin" || serviceType == "emby"
}

func (a VideoApps) Validate() error {
	if a.IOS != "" && a.IOS != "service" && a.IOS != "infuse" && a.IOS != "browser" {
		return fmt.Errorf("unsupported iPhone or iPad video app")
	}
	return nil
}

func (a VideoApps) WithDefaults(defaults *VideoApps) VideoApps {
	if a.IOS == "" && defaults != nil {
		a.IOS = defaults.IOS
	}
	if a.IOS == "" {
		a.IOS = "service"
	}
	return a
}

File: app/lib/features/media_access/data/video_apps.dart
import 'package:flutter/foundation.dart';

enum VideoApp {
  service('service'),
  infuse('infuse'),
  browser('browser');

  const VideoApp(this.id);
  final String id;

  static VideoApp fromId(String id) => values.firstWhere(
        (app) => app.id == id,
        orElse: () => service,
      );
}

/// Personal empty values inherit the instance default. Older servers and
/// unsupported platforms keep the service's existing app/browser behavior.
class VideoApps {
  const VideoApps({this.ios = ''});
  final String ios;

  static const serviceTypes = ['plex', 'jellyfin', 'emby'];

  factory VideoApps.fromJson(dynamic json) => VideoApps(
        ios: json is Map && json['ios'] is String ? json['ios'] as String : '',
      );

  Map<String, String> toJson() => {'ios': ios};

  VideoApp forPlatform(TargetPlatform platform, {required bool isWeb}) =>
      !isWeb && platform == TargetPlatform.iOS
          ? VideoApp.fromId(ios)
          : VideoApp.service;
}
